core/src/toga/documents.py

```diff
@@ -1,95 +1,66 @@
 from __future__ import annotations
 
-import asyncio
-import warnings
+import sys
 from abc import ABC, abstractmethod
+from collections.abc import Iterator
 from pathlib import Path
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, Mapping, Sequence
+
+import toga
+from toga import dialogs
+from toga.handlers import overridable, overridden
+from toga.window import MainWindow, Window
 
 if TYPE_CHECKING:
     from toga.app import App
-    from toga.window import Window
 
 
 class Document(ABC):
-    def __init__(
-        self,
-        path: str | Path,
-        document_type: str,
-        app: App,
-    ):
+    # Subclasses should override this definition
+
+    #: A short description of the type of document
+    document_type: str
```

Comment:
What do you think about renaming this variable to `description`? `type` suggests that its value should be a class, not a string.

Reply:
Agreed - and it also avoids the confusion between the app's `document_types` (which *is* a class), and this description.